commit 99d58e719acf0c5a2e41b59146e0e99afc51d6fb (HEAD -> main)
Author: Shin Morisawa <git@tryh4rd.dev>
Date:   Mon Mar 16 18:14:44 2026 +0900

    add my patches
    
    sss

diff --git a/config.def.h b/config.def.h
index be999b8..afe84fb 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,10 +6,10 @@
 /* appearance */
 static const int sloppyfocus               = 1;  /* focus follows mouse */
 static const int bypass_surface_visibility = 0;  /* 1 means idle inhibitors will disable idle tracking even if it's surface isn't visible  */
-static const unsigned int borderpx         = 1;  /* border pixel of windows */
+static const unsigned int borderpx         = 0;  /* border pixel of windows */
 static const float rootcolor[]             = COLOR(0x000000ff);
-static const float bordercolor[]           = COLOR(0x0e0e0eff);
-static const float focuscolor[]            = COLOR(0x20efefff);
+static const float bordercolor[]           = COLOR(0x000000ff);
+static const float focuscolor[]            = COLOR(0x000000ff);
 static const float urgentcolor[]           = COLOR(0xff0000ff);
 /* This conforms to the xdg-protocol. Set the alpha to zero to restore the old behavior */
 static const float fullscreen_bg[]         = {0.0f, 0.0f, 0.0f, 1.0f}; /* You can also use glsl colors */
@@ -43,7 +43,8 @@ static const MonitorRule monrules[] = {
    /* name        mfact  nmaster scale layout       rotate/reflect                x    y
     * example of a HiDPI laptop monitor:
     { "eDP-1",    0.5f,  1,      2,    &layouts[0], WL_OUTPUT_TRANSFORM_NORMAL,   -1,  -1 }, */
-	{ NULL,       0.55f, 1,      1,    &layouts[0], WL_OUTPUT_TRANSFORM_NORMAL,   -1,  -1 },
+	{ "HDMI-A-1", 0.55f, 1,      1,    &layouts[0], WL_OUTPUT_TRANSFORM_270,   -1,  -1 },
+	{ "DP-2",     0.55f, 1,      1,    &layouts[0], WL_OUTPUT_TRANSFORM_NORMAL,   -1,  -1 },
 	/* default monitor rule: can be changed but cannot be eliminated; at least one monitor rule must exist */
 };
 
@@ -56,7 +57,7 @@ static const struct xkb_rule_names xkb_rules = {
 	.options = NULL,
 };
 
-static const int repeat_rate = 25;
+static const int repeat_rate = 50;
 static const int repeat_delay = 600;
 
 /* Trackpad */
@@ -123,12 +124,12 @@ static const Key keys[] = {
 	/* modifier                  key                  function          argument */
 	{ MODKEY,                    XKB_KEY_m,           spawn,            {.v = menucmd} },
 	{ MODKEY,                    XKB_KEY_t,           spawn,            {.v = termcmd} },
-	{ MODKEY,                    XKB_KEY_j,           focusstack,       {.i = +1} },
-	{ MODKEY,                    XKB_KEY_k,           focusstack,       {.i = -1} },
+	{ MODKEY,                    XKB_KEY_h,           focusstack,       {.i = +1} },
+	{ MODKEY,                    XKB_KEY_l,           focusstack,       {.i = -1} },
 	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_I,           incnmaster,       {.i = +1} },
 	{ MODKEY,                    XKB_KEY_d,           incnmaster,       {.i = -1} },
-	{ MODKEY,                    XKB_KEY_h,           setmfact,         {.f = -0.05f} },
-	{ MODKEY,                    XKB_KEY_l,           setmfact,         {.f = +0.05f} },
+	{ MODKEY,                    XKB_KEY_u,           setmfact,         {.f = -0.05f} },
+	{ MODKEY,                    XKB_KEY_o,           setmfact,         {.f = +0.05f} },
 	{ MODKEY,                    XKB_KEY_Return,      zoom,             {0} },
 	{ MODKEY,                    XKB_KEY_Tab,         view,             {0} },
 	{ MODKEY,                    XKB_KEY_i,           killclient,       {0} },
@@ -136,10 +137,10 @@ static const Key keys[] = {
 	{ MODKEY,                    XKB_KEY_x,           setlayout,        {.v = &layouts[1]} },
 	{ MODKEY,                    XKB_KEY_c,           setlayout,        {.v = &layouts[2]} },
 	{ MODKEY,                    XKB_KEY_space,       setlayout,        {0} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_space,       togglefloating,   {0} },
+	{ MODKEY,                    XKB_KEY_v,           togglefloating,   {0} },
 	{ MODKEY,                    XKB_KEY_f,           togglefullscreen, {0} },
 	{ MODKEY,                    XKB_KEY_0,           view,             {.ui = ~0} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_parenright,  tag,              {.ui = ~0} },
+    { MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_parenright,  tag,              {.ui = ~0} },
 	{ MODKEY,                    XKB_KEY_comma,       focusmon,         {.i = WLR_DIRECTION_LEFT} },
 	{ MODKEY,                    XKB_KEY_period,      focusmon,         {.i = WLR_DIRECTION_RIGHT} },
 	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_less,        tagmon,           {.i = WLR_DIRECTION_LEFT} },
@@ -170,3 +171,54 @@ static const Button buttons[] = {
 	{ MODKEY, BTN_MIDDLE, togglefloating, {0} },
 	{ MODKEY, BTN_RIGHT,  moveresize,     {.ui = CurResize} },
 };
+
+static const Executables executables = {
+    (const Executable[]) {
+        {
+            .args = (const char*[]) { "/home/shin/.cargo/bin/mpd-discord-rpc", NULL }, .env = (const char*[]) { NULL }
+        },
+        {
+            .args = (const char*[]) { "/usr/bin/mako", NULL }, .env = (const char*[]) { NULL }
+        },
+        {
+            .args = (const char*[]) { "/usr/local/bin/mpdscribble", NULL }, .env = (const char*[]) { NULL }
+        },
+        {
+            .args = (const char*[]) { "/home/shin/.local/bin/fcitx5.sh", NULL }, .env = (const char*[]) { NULL }
+        },
+        {
+            .args = (const char*[]) { "/usr/bin/waybar", NULL }, .env = (const char*[]) { NULL }
+        }
+    },
+    5
+};
+
+static const Envvars env = {
+    (const Envvar[]) {
+        {
+            .name = "MOZ_ENABLE_WAYLAND", .value = "1"
+        },
+        {
+            .name = "MOZ_ACCELERATED", .value = "1"
+        },
+        {
+            .name = "QT_QPA_PLATFORM", .value = "wayland"
+        },
+        {
+            .name = "QT_IM_MODULES", .value = "wayland;fcitx;ibus"
+        },
+        {
+            .name = "OZONE_PLATFORM", .value = "wayland"
+        },
+        {
+            .name = "ELECTRON_OZONE_PLATFORM_HINT", .value = "auto"
+        },
+        {
+            .name = "XMODIFIERS", .value = "@im=fcitx"
+        },
+        {
+            .name = "GDK_BACKEND", .value = "wayland"
+        },
+    },
+    8
+};
